services/src/contexts/schema.sql

```diff
@@ -634,11 +634,96 @@ CREATE TABLE collection_children (
     PRIMARY KEY (parent, child)
 );
 
+CREATE TYPE "ArunaDataProviderDefinition" AS (
+    id uuid,
+    "name" text,
+    api_url text,
+    project_id text,
+    api_token text,
+    filter_label text,
+    cache_ttl int
+);
+
+CREATE TYPE "DatabaseConnectionConfig" AS (
+    host text,
+    port int,
+    "database" text,
+    schema text,
+    "user" text,
+    "password" text
+);
+
+CREATE TYPE "GbifDataProviderDefinition" AS (
+    "name" text,
+    db_config "DatabaseConnectionConfig",
+    cache_ttl int
+);
+
+CREATE TYPE "GfbioAbcdDataProviderDefinition" AS (
+    "name" text,
+    db_config "DatabaseConnectionConfig",
+    cache_ttl int
+);
+
+CREATE TYPE "GfbioCollectionsDataProviderDefinition" AS (
+    "name" text,
+    collection_api_url text,
+    collection_api_auth_token text,
+    abcd_db_config "DatabaseConnectionConfig",
+    pangaea_url text,
+    cache_ttl int
+);
+
+CREATE TYPE "TextMetaDataDefinitionKeyValue" AS (
+    key text,
+    value "MetaDataDefinition"
+);
+
+CREATE DOMAIN "LayerId" AS text;
+
+CREATE DOMAIN "LayerCollectionId" AS text;
+
+
```

Comment:
are the newlines on purpose?

Reply:
[5c7aaa7](https://github.com/geo-engine/geoengine/pull/866/commits/5c7aaa7bdbd00967ec3bff7f8e786518c9360903)